src/OpenTelemetry.Extensions.Propagators/JaegerPropagator.cs

```diff
@@ -0,0 +1,180 @@
+// <copyright file="JaegerPropagator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Extensions.Propagators;
+
+/// <summary>
+/// A text map propagator for Jaeger trace context. See https://www.jaegertracing.io/docs/next-release/client-libraries/#propagation-format.
+/// </summary>
+public class JaegerPropagator : TextMapPropagator
+{
+    internal const string JaegerHeader = "uber-trace-id";
+    internal const string JaegerDelimiter = ":";
+    internal const string JaegerDelimiterEncoded = "%3A"; // while the spec defines the delimiter as a ':', some clients will url encode headers.
+    internal const string SampledValue = "1";
+
+    internal static readonly string[] JaegerDelimiters = { JaegerDelimiter, JaegerDelimiterEncoded };
+
+    private static readonly int TraceId128BitLength = "0af7651916cd43dd8448eb211c80319c".Length;
+    private static readonly int SpanIdLength = "00f067aa0ba902b7".Length;
+
+    /// <inheritdoc/>
+    public override ISet<string> Fields => new HashSet<string> { JaegerHeader };
+
+    /// <inheritdoc/>
+    public override PropagationContext Extract<T>(PropagationContext context, T carrier, Func<T, string, IEnumerable<string>> getter)
+    {
+        if (context.ActivityContext.IsValid())
+        {
+            // If a valid context has already been extracted, perform a noop.
+            return context;
+        }
+
+        if (carrier == null)
+        {
+            OpenTelemetryApiEventSource.Log.FailedToExtractActivityContext(nameof(JaegerPropagator), "null carrier");
```

Comment:
we want to start having own eventsource for this new package.

Reply:
@cijothomas , can we punt on this?? i'm leaving the company that this PR is associated to this week.

Reply:
To ship the stable version of this, I fixed this to use own eventsource. You need to pick the latest from main to see the changes reflected.